dimagi/utils/django/cached_object.py

```diff
@@ -271,6 +296,83 @@ def get_all_keys(self):
         return full_stream_keys, full_meta_keys
 
 
+class FileObject(TemporaryObject):
+    """
+    Leverages NFS filesystem for caching large documents
+    """
+    def __init__(self, cache_key, base_dir=None):
+        self._base_dir = base_dir
+        super(FileObject, self).__init__(cache_key)
+
+    def fetch_stream(self):
+        stream_path = os.path.join(
+            self.base_dir,
+            self.stream_key(),
+        )
+        try:
+            return open(stream_path, 'r+')
+        except IOError:
+            return None
+
+    def fetch_meta(self):
+        meta_path = os.path.join(
+            self.base_dir,
+            self.meta_key(),
+        )
+        with open(meta_path, 'r+') as f:
+            return simplejson.loads(f.read())
+        return None
+
+    def get(self, **kwargs):
+        stream = self.fetch_stream()
+        meta = self.fetch_meta()
+
+        return (meta, stream)
+
+    def get_stream_filepath(self):
+        return os.path.join(self.base_dir, self.stream_key())
+
+    def get_meta_filepath(self):
+        return os.path.join(self.base_dir, self.meta_key())
+
+    @property
+    def base_dir(self):
+        return self._base_dir or settings.SHARED_DRIVE_CONF.temp_dir
+
+    def cache_put(self, object_stream, metadata, timeout=None):
+        object_meta = CachedObjectMeta.make_meta(object_stream, OBJECT_ORIGINAL, metadata)
+
+        object_stream.seek(0)
+
+        base_dir = self.base_dir
+        # Do not cache if there is no base dir
+        if not base_dir:
+            return
+
+        stream_path = self.get_stream_filepath()
+        meta_path = self.get_meta_filepath()
+
+        with open(stream_path, 'w+') as f:
+            f.write(object_stream.read())
+        with open(meta_path, 'w+') as f:
+            f.write(simplejson.dumps(object_meta.to_json()))
+
+    def get_all_keys(self):
+        """
+        Returns all FULL keys
+        """
+        if not self.base_dir:
+            return [], []
+        full_stream_keys = glob.glob(os.path.join(self.base_dir, self.stream_key(WILDCARD)))
+        full_meta_keys = glob.glob(os.path.join(self.base_dir, self.meta_key(WILDCARD)))
```

Comment:
yea i agree. i actually think there will only ever be one cache key. the sizes are used for `CachedImage`. so i think we can minimally just call `os.path.exists(stream_key)` and that should tell us the correct information